evm/contracts/apps/ucs/03-zkgm/Zkgm.sol

```diff
@@ -49,69 +51,314 @@ contract AbiExport {
     ) public {}
 }
 
-contract UCS03Zkgm is
-    IBCAppBase,
-    Initializable,
-    UUPSUpgradeable,
+function passthrough(
+    address impl
+) {
+    assembly {
+        calldatacopy(0, 0, calldatasize())
+        let result := delegatecall(gas(), impl, 0, calldatasize(), 0, 0)
+        returndatacopy(0, 0, returndatasize())
+        switch result
+        case 0 { revert(0, returndatasize()) }
+        default { return(0, returndatasize()) }
+    }
+}
+
+abstract contract UCS03ZkgmStore is AccessManagedUpgradeable, IZkgmStore {
+    bytes32 internal constant STAKE_NFT_MANAGER_SALT =
+        keccak256("union.salt.zkgm.stakeNFTManager");
+
+    string internal constant STAKE_NFT_NAME = "Zkgm Staking Position";
+    string internal constant STAKE_NFT_SYMBOL = "ZKGMSP";
```

Comment:
did we settle on naming for the zkgm positions?

Reply:
I've opened on linear